docs: container use blog and guide (#2962)
block/goose: documentation/blog/2025-06-17-isolated-development-environments/index.md

---

> title: "Isolated Dev Environments in Goose with conainer-use"

Reviewer: ```suggestion
title: "Isolated Dev Environments in Goose with container-use"
```

> description: cNever worry about breaking your development setup again with containerized, git-branch-isolated development environments powered by container-use

Reviewer: ```suggestion
description: Never worry about breaking your development setup again with containerized, git-branch-isolated development environments powered by container-use
```

authors:
    - mic
tags: [extensions, containers, development, isolation, mcp, dagger]
---

> ![Dagger Logo](https://avatars.githubusercontent.com/u/78824383?v=4&s=100)

Reviewer 2: Just updated our GitHub avatar to be higher resolution. If you prefer I can share the high res file directly, but this is fine

Author: I think this one looks ok, it is pretty small - or is there a CDN link?

Reviewer 2: I think it's the `&s=100` making it small, without it it's fairly large.

No CDN link unfortunately, we have PNG/SVGs in a public GDrive.

Reviewer 3: https://avatars.githubusercontent.com/u/78824383?v=4

> Over 10 years ago Docker came on the scene and introduced masses of developers to the concept and practice of containers.

Reviewer: ```suggestion
Over ten years ago, Docker came onto the scene and introduced developers en masse to the concept and practice of containers.
```

> These containers were useful to solve deployment and buildtime problems (and in some cases development environments) and rapidly became mainstream. 

Reviewer: ```suggestion
These containers helped solve deployment and build-time problems, and in some cases, issues with development environments. They quickly became mainstream.

```

Underlying these containers was interesting tech around "copy on write" filesystems and isolated "not quite" virtual-machine-like environments (certainly helped to contain processes and clean up after).

> Dagger (the project and company!) followed on from Docker (by [Solomon Hykes](https://www.linkedin.com/in/solomonhykes/), creator of Docker) and furthered the reach of containers for developers.

Reviewer: ```suggestion
Dagger, the project and company founded by Docker’s creator [Solomon Hykes](https://www.linkedin.com/in/solomonhykes), has furthered the reach of containers for developers.
```

> A project that has come out of that is an MCP server called `container-use` - which provides a cli and tool which can be invaluable for coding agents to able to do work in isolated environments and branches with a clear lifecycle and ability to roll back, limit changes and risk (compared to running an agent direct on your system) but with the same ergonomics and convenience you are used to with agents. 

Reviewer: ```suggestion
A project that has come out of that is an MCP server called `container-use` - which provides a cli and tool which can be invaluable for coding agents to able to do work in isolated environments and branches with a clear lifecycle and ability to roll back, limit changes and risk (compared to running an agent directly on your system) but with the same ergonomics and convenience you are used to with agents. 
```
